### PR TITLE
GH-41385: [CI][MATLAB][Packaging] Add support for MATLAB `R2024a` in CI and crossbow packaging workflows

### PR DESCRIPTION
**Pull request by sgilmore10**
### Rationale for this change

MATLAB `R2024a` is the latest available version of MATLAB as of April 2024.

We are currently building against MATLAB `R2023a` in CI and for the crossbow packaging workflow.

We should update the version of MATLAB we support to the latest available version.

We previously created an issue to use `R2023b` (https://github.com/apache/arrow/issues/37809). However, `R2024a` has become publicly available since then.

### What changes are included in this PR?

1. Changed the `release` argument supplied to `matlab-actions/setup-matlab@v2` to `R2024a` from `R2023a` in `.github/workflows/matlab.yml` and `dev/tasks/matlab/github.yml`.

2. Updated the script used to package the MLTBX file (`arrow/matlab/tools/packageMatlabInterface.m`) to specify the version of MATLAB currently running as the only compatible version of MATLAB for the interface. 

3.  Updated display tests to react to a change in how class names are formatted when an objected is displayed in the Command Window.

### Are these changes tested?

Yes. Existing tests used.

### Are there any user-facing changes?

There are no changes to the MATLAB Arrow Interface. However, the MATLAB release used to build, test, and package the interface has been upgraded.

### Future Work

1. We may want to consider building and packaging against multiple versions of MATLAB in parallel, rather than just the latest. This would require some more thought on how many releases back to support over time, however.

2. #41435

* GitHub Issue: #41385

### CODE REVIEW

**Comment by sgilmore10** [comment]
@github-actions crossbow submit matlab

**Comment by sgilmore10** [comment]
@github-actions crossbow submit matlab

**Comment by sgilmore10** [comment]
Verified the packaged MATLAB Arrow Interface manually by

1. Installing the MLTBX file on `Debian-11`, `macos-12`, `macos-14`, and `Windows 10` AND
2. Running all unit tests underneath `arrow/matlab/test`

File: matlab/tools/packageMatlabInterface.m [hunk 1]
```diff
@@ -55,9 +55,18 @@
...
 
-% Interface is only qualified against R2023a at the moment
-opts.MinimumMatlabRelease = "R2023a";
-opts.MaximumMatlabRelease = "R2023a";
+% MEX files use run-time libraries shipped with MATLAB (e.g. libmx, libmex,
+% etc.). MEX files linked against earlier versions of MALTAB run-time libraries
+% will most likely work on newer versions of MATLAB. However, this may not
+% always be the case.
+% 
+% For now, set the earliest and latest compatible  releases of MATLAB to
```
Comment (kou) [inline matlab/tools/packageMatlabInterface.m:63]:
```suggestion
% For now, set the earliest and latest compatible releases of MATLAB to 
```

**Review by kou — APPROVED** [review]
+1

**Review by raulcd — APPROVED** [review]
LGTM

**Pull request merged**